### PR TITLE
ref(ts): Convert <Resolve> to typescript

### PR DESCRIPTION
**Pull request by matejminar**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/actions/resolve.tsx [hunk 1]
```diff
@@ -9,8 +9,42 @@ import ActionLink from 'app/components/actions/actionLink';
...
+
+const defaultProps = {
+  isResolved: false,
+  isAutoResolved: false,
+  confirmLabel: t('Resolve'),
+};
+
+type Props = {
+  hasRelease: boolean;
+  onUpdate: (params: {status: ResolveStatus; statusDetails?: StatusDetails}) => void;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/actions/resolve.tsx]: Should this be a type that is exported? Will parent components need this type too?

Comment (matejminar) [reply src/sentry/static/sentry/app/components/actions/resolve.tsx]: Yes, will be handy to have it exported when converting `actions.jsx`, thanks

File: src/sentry/static/sentry/app/types/index.tsx [hunk 2]
```diff
@@ -866,3 +866,16 @@ export type Chunks = {
   remark?: string;
   rule_id?: string;
 };
+
+export enum ResolvementStatus {
```
Comment (markstory) [inline src/sentry/static/sentry/app/types/index.tsx]: ResolutionStatus might be a better name. Resolvement isn't a standard word.

**Review by markstory — APPROVED** [review]
Looks good other than the Resolvement name. I would use Resolution instead.

File: src/sentry/static/sentry/app/types/index.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/types/index.tsx]: Thanks!

tenor

**Pull request merged**